Converters/Revit/Speckle.Converters.RevitShared/ToSpeckle/Properties/PropertiesExtractor.cs

```diff
@@ -0,0 +1,56 @@
+using Microsoft.Extensions.Logging;
+using Speckle.Converters.Common.Objects;
+using Speckle.Sdk;
+
+namespace Speckle.Converters.RevitShared.ToSpeckle.Properties;
+
+public class PropertiesExtractor
+{
+  private readonly ParameterExtractor _parameterExtractor;
+  private readonly ITypedConverter<DB.Element, Dictionary<string, object>> _materialQuantityConverter;
+  private readonly ILogger<RevitRootToSpeckleConverter> _logger;
+
+  public PropertiesExtractor(
+    ParameterExtractor parameterExtractor,
+    ITypedConverter<DB.Element, Dictionary<string, object>> materialQuantityConverter,
+    ILogger<RevitRootToSpeckleConverter> logger
+  )
+  {
+    _parameterExtractor = parameterExtractor;
+    _materialQuantityConverter = materialQuantityConverter;
+    _logger = logger;
+  }
+
+  public Dictionary<string, object?> GetProperties(DB.Element element)
+  {
+    Dictionary<string, object?> properties = new();
+
+    try
+    {
+      Dictionary<string, object> matQuantities = _materialQuantityConverter.Convert(element);
+      if (matQuantities.Count > 0)
+      {
+        properties.Add("Material Quantities", matQuantities);
+      }
+    }
+    catch (Exception e) when (!e.IsFatal())
```

Comment:
With V3, we wanted to er on the side of caution with these silent failures.

With this code right now, the user will see zero warning.
I would prefer us to let these exceptions bubble up, and do some testing to see the cases where they occur.

Reply:
done

Reply:
I would put here a note `TODO: revisit this once we have better reporting on "converted with warnings"`. regardless not sure how/when likely this can happen?